Projects/UOContent/Gumps/ObjectViewer/RegionView.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Server.Regions;
+using Microsoft.Extensions.DependencyInjection;
+using static Server.DataModel;
+
+namespace Server.Gumps
+{
+    public class RegionView : BaseViewGump<RegionModel>
+    {
+        private IRegionStore RegionStore;
+        public override bool TwoButton => true;
+        
+        public override void OnLoad()
+        {
+            RegionStore = DataStore.ServiceProvider.GetService<IRegionStore>();
+            Collection = RegionStore.GetStore()?.ToList();
+        }
+        public RegionView(Mobile owner, int page = 0) : base(owner, page) { }
+       
+        public override void OnAdd(Mobile m, string name)
+        {
+            if (name.Length < 2)
+            {
+                m.SendMessage("The name cannot be less than 2 characters");
+                return;
+            }
+            if (RegionStore.Exist(name))
+            {
+                m.SendMessage("Name is already used");
+                return;
+            }
+
+            RegionStore.Add(name, new RegionModel() { Name = name });
+            ReOpen();
+        }
+
+        public override void OnPressTwoButton(Mobile m, int index)
+        {
+            if (Collection.ElementAt(index).Value is RegionModel region)
```

Comment:
Is this always true?, I assume `Collection` is `List<KeyValuePair<string, RegionModel>>`.
```cs
var region = Collection[index].Value;
```